src/Microsoft.DncEng.Configuration.Extensions/MappedJsonConfigurationProvider.cs

```diff
@@ -33,8 +42,21 @@ private void MapData()
 
         private void Reload(object? state)
         {
-            MapData();
-            OnReload();
+            try
+            {
+                MapData();
+                OnReload();
+            }
+            catch (Exception e)
+            {
+                // This exception, because it's in a System.Threading.Timer
+                // is going to crash the process, if we are reporting to AppInsights, let's send it to the channel
+                // and flush it, so that it gets recorded somewhere
+                _telemetry?.TrackException(e);
+                _telemetry?.Flush();
```

Comment:
We have an IServiceProvider available.  Are there multiple of those? The entire IConfiguration section is in a AddSinglton, so it should have all the stuff it needs.  Though maybe it's going to cause a cycle (surely the TelemetryClient stuff needs the settings).

Reply:
I was thinking of a previous implementation. Ignore me.

Reply:
Discussed this with Alex, and it seems like we've centralized things a bit since this happened.  I _did_ move the resolution to the Timer callback, so it's not a constructor dependency loop (since TelemetryClient is DEFINITELY going to need config to be set up correctly).  That should let it work